### PR TITLE
[ProteinSolvation] Add maxwarn flag to grompp command

### PR DESCRIPTION
**Pull request by cristianacarpinteiro**
The addition of the maxwarn flag to the grompp command prevents experiments to fail for this common errors: 
```
 WARNING 1 [file energy_minimization.mdp]:
  The largest distance between excluded atoms is 1.208 nm, which is larger
  than the cut-off distance. This will lead to missing long-range
  corrections in the forces and energies. If you expect that minimization
  will bring such distances within the cut-off, you can ignore this warning.
```
According to [this](https://gromacs.bioexcel.eu/t/the-largest-distance-between-excluded-atoms-is-1-725-nm-which-is-larger-than-the-cut-off-distance/4153/2) the user should take a look to check if the protein structure is correct, however, the problem can be overriden by setting the `maxwarn` flag to 1.
I also remove the ns_type variable, since it was already the default value.

### CODE REVIEW

**Review by david451lima — APPROVED** [review]
LGTM

**Comment by IvanPombo** [comment]
@cristianacarpinteiro according to this (and following the structures of other simulators) it means that if more than one warning appears, then the simulation crashes. Is this correct?

Moreover, this error won't be passed to the user until the end of the simulation, so we won't be able to stop it if he did not want this happening. So the simulation can work, but users may still want to kill it for these warnings. No? If so, we need to start thinking about how to send this to the user.

**Comment by cristianacarpinteiro** [comment]
> @cristianacarpinteiro according to this (and following the structures of other simulators) it means that if more than one warning appears, then the simulation crashes. Is this correct?
> 
> Moreover, this error won't be passed to the user until the end of the simulation, so we won't be able to stop it if he did not want this happening. So the simulation can work, but users may still want to kill it for these warnings. No? If so, we need to start thinking about how to send this to the user.

Yes, you are correct! These warnings appear in a lot of RCSB files and are very specific to each protein to solve in a systematic way. Ignoring this warnings will produce a simulation with some minor inaccuracies according to the documentation, but I think it should be up to the user if he wants to ignore the warnings or not. What is your suggestion on how to pass this to the user?

**Comment by IvanPombo** [comment]
> Yes, you are correct! These warnings appear in a lot of RCSB files and are very specific to each protein to solve in a systematic way. Ignoring this warnings will produce a simulation with some minor inaccuracies according to the documentation, but I think it should be up to the user if he wants to ignore the warnings or not. What is your suggestion on how to pass this to the user?

I do not have a suggestion as of yet, but at least for warnings there should be a certain stream from the backend so that users can immediately address them if they want. Otherwise, they consume computation and only at the end they see their simulation was useless.

**Comment by cristianacarpinteiro** [comment]
I think the best way of approaching this as of now is to pass a ignore_warnings in the simulate method. If the user sets it to True we add the maxwarn flag and if it sets to False the simulation will fail and then the user can see the warnings on the stdout. Streaming the warnings to the user directly seems a too complicated solution to implement quickly.

**Comment by david451lima** [comment]
> simulate method. If the user sets it to True we add the maxwarn flag and if it sets to False the simulation will fail and then the user can see the warnings on the stdout. Streaming the warnings to the user directly seems a too complicated solutio

Seems a reasonable solution. The user can check immediatly in the stdout if the simulation crashed due to warnings and pass the argument accordingly

**Comment by cristianacarpinteiro** [comment]
> > I think the best way of approaching this as of now is to pass a ignore_warnings in the simulate method. If the user sets it to True we add the maxwarn flag and if it sets to False the simulation will fail and then the user can see the warnings on the stdout. Streaming the warnings to the user directly seems a too complicated solution to implement quickly.
> 
> I didn't mention to do it quickly :) Just to start thinking about it!
> 
> Okay, that seems like a reasonable solution! However, with `-maxwarn 1` this will just ignore the first warning and crash on the second. What do you think you should provide? Ignore all? A few? Let the user decide?

I will put it as a value the user can change instead of a bool.

**Comment by cristianacarpinteiro** [comment]
@david451lima @IvanPombo I am re-requesting your review because I changed the way I was doing this to address our concerns.

File: inductiva/molecules/protein_solvation/protein_solvation.py [hunk 1]
```diff
@@ -45,7 +46,8 @@ def simulate(
             simulation_time_ns: float = 10,  # ns
             output_timestep_ps: float = 1,  # ps
             integrator: Literal["md", "sd", "bd"] = "md",
-            n_steps_min: int = 5000) -> tasks.Task:
+            n_steps_min: int = 5000,
+            ingore_warnings: bool = False) -> tasks.Task:
```
Comment (IvanPombo) [inline inductiva/molecules/protein_solvation/protein_solvation.py:50]: Typo

File: inductiva/molecules/protein_solvation/protein_solvation.py [hunk 2]
```diff
@@ -78,6 +82,10 @@ def simulate(
         # of the simulation (2 fs)
         self.integrator = integrator
         self.n_steps_min = n_steps_min
+        self.ignore_warnings = 0
+        if ingore_warnings:
```
Comment (IvanPombo) [inline inductiva/molecules/protein_solvation/protein_solvation.py:86]: typo

File: inductiva/molecules/protein_solvation/protein_solvation.py [hunk 3]
```diff
@@ -78,6 +82,10 @@ def simulate(
...
+            self.ignore_warnings = -1
```
Comment (IvanPombo) [inline inductiva/molecules/protein_solvation/protein_solvation.py:87]: Why -1?

Comment (IvanPombo) [reply inductiva/molecules/protein_solvation/protein_solvation.py:87]: Is to catch all?

**Review by IvanPombo — APPROVED** [review]
Minor comments. LGTM!

File: inductiva/molecules/protein_solvation/protein_solvation.py (hunk 3 above)
Comment (cristianacarpinteiro) [reply inductiva/molecules/protein_solvation/protein_solvation.py:87]: yes, all of them are ignored.

Comment (IvanPombo) [reply inductiva/molecules/protein_solvation/protein_solvation.py:87]: Cool, thanks!

**Pull request merged**